### DISCUSSION TITLE
Network library not found in micropython 1.20

### DISCUSSION THREAD

**Discussion by dirkdemuynck** [micropython/micropython#11375] [RP2040 / Pico] [answered]
Just installed micropython 1.20 in rpi pico w. When trying to import the network library, it fails. I had to go back to the specific 1.19 version for pico, to have networking operational.
However, I understood that pico w should be supported in 1.20. but apparently not.

**Comment by robert-hh** [comment]
What do you get when you push Ctrl-B in REPL? I get for the v1.20.0:
```
MicroPython v1.20.0 on 2023-04-26; Raspberry Pi Pico W with RP2040
Type "help()" for more information.
```
And the network module is present.

**Comment by dirkdemuynck** [comment]
I get : 
_MicroPython v1.20.0 on 2023-04-26; Raspberry Pi Pico with RP2040_
So it sees it as a rpi pico. However, it is a pico W, as I can see stamped on the microcontroller.
How is this possible ?

**Reply by robert-hh** [reply]
You uploaded the wrong firmware

**Comment by dirkdemuynck** [comment]
This is the same hardware with 1.19. It works fine with that one
MicroPython v1.19.1-1019-g9e6885ad8 on 2023-04-22; Raspberry Pi Pico W with RP2040

For 1.20, I uploaded this one : https://micropython.org/download/rp2-pico-w/
That should be correct . Where do I find the other firmware?

**Reply by robert-hh** [reply, marked as answer]
https://micropython.org/resources/firmware/rp2-pico-w-20230426-v1.20.0.uf2

That's the one which I just loaded for verification. Just try it again with that file. The firmware file has to include pico-w in it's name.

**Comment by dirkdemuynck** [comment]
Thanks. It works now !
But it also means that the page https://micropython.org/download/rp2-pico-w/
contains the wrong firmware because it says Pico W. I don't know who maintains that page.

**Reply by robert-hh** [reply]
For me the page looks right, because I got the link from that page. What was the name of the file you downloaded initially?

**Comment by dirkdemuynck** [comment]
You're right. I must have been mistaking.
Thanks for your help.
